zebra/zebra_vrf.c

```diff
@@ -339,11 +339,6 @@ static int zebra_vrf_delete(struct vrf *vrf)
 	otable_fini(&zvrf->other_tables);
 	XFREE(MTYPE_ZEBRA_VRF, zvrf);
 
-	if (vrf->ns_ctxt) {
-		ns_delete(vrf->ns_ctxt);
-		vrf->ns_ctxt = NULL;
-	}
```

Comment:
isn't this introducing a leak when using the netns vrf backend?

Reply:
No it is not because it is done later. It was just too early.

There are two situations that are described in the commit log

- If a netns is removed from the system, vrf_delete()->zebra_vrf_delete()
  calls ns_delete(). See
https://github.com/FRRouting/frr/blob/593ad76f993626de9b7d807312978477e1335460/zebra/zebra_vrf.c#L343
- If zebra is terminating, zebra_ns_final_shutdown() will call
  zebra_vrf_delete() which call ns_delete(). See https://github.com/FRRouting/frr/blob/593ad76f993626de9b7d807312978477e1335460/zebra/zebra_ns.c#L400

Reply:
Ha, sorry for missing this in the commit log. Thanks for the clarification!